Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenUpdaterSystem.cs

```diff
@@ -0,0 +1,220 @@
+                sdkTweenComponent.isPlaying = isPlaying;
+
+                Transform entityTransform = transformComponent.Transform;
+                float durationInSeconds = tweenModel.Duration / 1000;
+
+                currentTweener = sdkTweenComponent.tweener;
+
+                // There may be a tween running for the entity transform, even though internalComponentModel.tweener
+                // is null, e.g: during preview mode hot-reload.
+                List<DG.Tweening.Tween> transformTweens = DOTween.TweensByTarget(entityTransform, true);
+                transformTweens?[0].Rewind(false);
+
+                if (!EASING_FUNCTIONS_MAP.TryGetValue(tweenModel.EasingFunction, out Ease ease))
+                    ease = Linear;
+
+                switch (tweenModel.ModeCase)
+                {
+                    case PBTween.ModeOneofCase.Rotate:
+                        currentTweener = SetupRotationTween(transformComponent,
+                            PBQuaternionToUnityQuaternion(tweenModel.Rotate.Start),
+                            PBQuaternionToUnityQuaternion(tweenModel.Rotate.End),
+                            durationInSeconds, ease);
+
+                        break;
+                    case PBTween.ModeOneofCase.Scale:
+                        currentTweener = SetupScaleTween(ref transformComponent,
+                            PBVectorToUnityVector(tweenModel.Scale.Start),
+                            PBVectorToUnityVector(tweenModel.Scale.End),
+                            durationInSeconds, ease);
+
+                        break;
+                    case PBTween.ModeOneofCase.Move:
+                    default:
+                        currentTweener = SetupPositionTween(transformComponent,
+                            PBVectorToUnityVector(tweenModel.Move.Start),
+                            PBVectorToUnityVector(tweenModel.Move.End),
+                            durationInSeconds, ease, tweenModel.Move.HasFaceDirection && tweenModel.Move.FaceDirection);
+
+                        break;
+                }
+
+                currentTweener.Goto(tweenModel.CurrentTime * durationInSeconds, isPlaying);
+
+                sdkTweenComponent.tweener = currentTweener;
+                sdkTweenComponent.CurrentTime = tweenModel.CurrentTime;
+                sdkTweenComponent.IsDirty = false;
+
+                if (isPlaying)
+                {
+                    sdkTweenComponent.tweener.Play();
+                    sdkTweenComponent.TweenStateStatus = sdkTweenComponent.CurrentTime.Equals(1f) ? TweenStateStatus.TsCompleted : TweenStateStatus.TsActive;
+                }
+                else
+                {
+                    sdkTweenComponent.tweener.Pause();
+                    sdkTweenComponent.TweenStateStatus = TweenStateStatus.TsPaused;
+                }
+            }
+        }
+
+        public static Vector3 PBVectorToUnityVector(Decentraland.Common.Vector3 original) =>
```

Comment:
Let's move these two methods to some common extensions. E.g. similarly to `PrimitivesInitializationExtensions`

Reply:
moved it to a static `PrimitivesConversionExtensions`, I'm not 100% sure about the naming conventions yet, so it can be changed of course